RecoTracker/MkFitCore/src/CandCloner.cc

```diff
@@ -0,0 +1,169 @@
+          while (extra_i != extra_e && sortByScoreTrackCand(*extra_i, tc) &&
+                 n_pushed < mp_iteration_params->maxCandsPerSeed) {
+            cv.emplace_back(*extra_i);
+            ++n_pushed;
+            ++extra_i;
+          }
+
+          if (n_pushed >= mp_iteration_params->maxCandsPerSeed)
+            break;
+
+          // set the overlap if we have a true hit and pT > pTCutOverlap
+          HitMatch *hm;
+          if (tc.pT() > mp_iteration_params->pTCutOverlap && h2a.hitIdx >= 0 &&
+              (hm = ccand[h2a.trkIdx].findOverlap(h2a.hitIdx, h2a.module))) {
+            tc.addHitIdx(hm->m_hit_idx, m_layer, hm->m_chi2);
+            tc.incOverlapCount();
+
+            // --- ROOT text tree dump of all used overlaps
+            // static bool first = true;
+            // if (first)
+            // {
+            //   // ./mkFit ... | perl -ne 'if (/^ZZZ_EXTRA/) { s/^ZZZ_EXTRA //og; print; }' > extra.rtt
+            //   printf("ZZZ_EXTRA label/I:can_idx/I:layer/I:pt/F:eta/F:phi/F:"
+            //          "chi2_orig/F:chi2/F:chi2_extra/F:module/I:module_extra/I\n");
+            //   first = false;
+            // }
+            // label/I:can_idx/I:layer/I:pt/F:eta/F:phi/F:chi2_orig/F:chi2/F:chi2_extra/F:module/I:module_extra/I
+            // printf("ZZZ_EXTRA %d %d %d %f %f %f %f %f %f %u %u\n",
+            //        tc.label(), h2a.trkIdx, m_layer, tc.pT(), tc.posEta(), tc.posPhi(),
+            //        ccand[h2a.trkIdx].chi2(), h2a.chi2, hm->m_chi2, h2a.module, hm->m_module_id);
+          }
+
+          cv.emplace_back(tc);
+          ++n_pushed;
+
+          if (h2a.hitIdx >= 0) {
+            mp_kalman_update_list->push_back(std::pair<int, int>(m_start_seed + is, n_pushed - 1));
+          }
+        }
+
+        // Add remaining extras as long as there is still room for them.
+        while (extra_i != extra_e && n_pushed < mp_iteration_params->maxCandsPerSeed) {
+          cv.emplace_back(*extra_i);
+          ++n_pushed;
+          ++extra_i;
+        }
+
+        //ccand.swap(cv); // segfaulting w/ backwards fit input tracks -- using loop below now
+        ccand.resize(cv.size());
+        for (size_t ii = 0; ii < cv.size(); ++ii) {
+          ccand[ii] = cv[ii];
+        }
+        cv.clear();
+      } else  // hitsForSeed.empty()
+      {
+        if (ccand.m_state == CombCandidate::Finding) {
+          ccand.clear();
+
+          while (extra_i != extra_e) {
+            ccand.emplace_back(*extra_i);
+            ++extra_i;
+          }
+        }
+
+        // Cross-check for what is left once there are no more changes for a whole seed.
+        //
+        // for (auto &cand : cands[ m_start_seed + is ])
+        // {
+        //   if (cand.getLastHitIdx() < 0 && cand.getLastHitIdx() != -2)
+        //   {
+        //     printf("  We got cand with index %d\n", cand.getLastHitIdx());
+        //     print("offending track (unknown index)", -666, cand, true);
+        //   }
+        // }
```

Comment:
still needed?

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit: https://github.com/trackreco/cmssw/pull/76/commits/842cdaf057f74ecf568f12bb1a4cb96e19c5d22c